wallet/dapps-communication.md

```diff
@@ -427,284 +428,41 @@ authentication, and authorization to prevent unauthorized access or data breache
+const publicKey = "Pxxxxxxxxxxxxxxx";
+await myProvider.importAccount(publicKey, privateKey);
 
-// Register
-window.massaWalletProvider.addEventListener('register', (payload) => {
-  const extensionEventTarget = new EventTarget();
-  window[`massaWalletProvider-${payload.eventTarget}`] = extensionEventTarget;
-  registeredProviders[payload.providerName] = payload.eventTarget;
-})
+// Get accounts.
+const myAccounts = await myProvider.accounts();
 
-interface Message {
-  params: object;
-  requestId: string;
-}
-
-const availableCommands = ['ListAccounts', 'DeleteAccount', 'ImportAccount', 'Balance', 'Sign']
-
-const pendingRequests = new Map<string, Function>();
-
-//send a message from the webpage script to the content script
-function sendMessageToContentScript(provider, command, params, responseCallback) {
-  const requestId = uuidv4();
-  const message: Message = { params, requestId };
-  pendingRequests.set(requestId, responseCallback);
-
-  if (!availableCommands.includes(command)) throw new Error('Unhandled command)
-
-  window[`massaWalletProvider-${registeredProviders[provider]}`].dispatchEvent(
-    new CustomEvent(command, { detail: message })
-    );
-}
-
-//receive a response from the content script
-function handleResponseFromContentScript(event) {
-  const { result, error, requestId } = event.detail;
-  const responseCallback = pendingRequests.get(requestId);
-
-  if (responseCallback) {
-    if (error) {
-      responseCallback(new Error(error));
-    } else {
-      responseCallback(null, result);
-    }
-    pendingRequests.delete(requestId);
-  }
-}
-
-window.massaWalletProvider.addEventListener('message', handleResponseFromContentScript);
-```
-
-Because it was requested, here a potential implementation of the `registerAsMassaWalletProvider` and `sign` in the
-content script:
+// Get one account.
+const myAccount = myAccounts[0];
 
-```typescript
-interface RegistrationPayload {
-  providerName: string;
-  eventTarget: string;
-}
-
-function registerAsMassaWalletProvider(providerName: string): Promise<boolean> {
-  return new Promise((resolve) => {
-    const registerProvider = () => {
-      window.massaWalletProvider.dispatchEvent(
-        new CustomEvent('register', {
-          detail: { providerName: providerName, eventTarget: providerName } as RegistrationPayload
-        })
-      );
-      resolve(true);
-    };
-
-    if (document.readyState === 'complete' || document.readyState === 'interactive') {
-      registerProvider();
-    } else {
-      document.addEventListener('DOMContentLoaded', registerProvider);
-    }
-  });
-}
+// Get the account's address.
+const accountAddress = myAccount.address();
 
-const actionToCallback = new Map<string, Function>()
+// Get the account balance.
+const accountBalance = await myAccount.balance();
 
-function sign(callback: Function): void {
-  actionToCallback.set('sign', callback);
-}
+// Sign a message.
+const signature = await myAccount.sign([0, 1, 2]);
```

Comment:
No blocking, just asking, what is the function in massa-web3 side to generate the payload in byte that we want to sign? 

Should we reference this function somewhere here?